CellIoT/gsm_at_lib/src/gsm/gsm_int.c

```diff
@@ -1057,6 +1065,10 @@ gsmi_parse_received(gsm_recv_t* rcv) {
             gsmi_parse_ip(&tmp, &gsm.m.network.ip_addr);/* Parse IP address */
 
             is_ok = 1;                          /* Manually set OK flag as we don't expect OK in CIFSR command */
+        } else if (!strncmp(rcv->data, "<<<", 3)) {
+            strncpy(gsm.m.http_response.data, &rcv->data[3], sizeof(gsm.m.http_response.data));
+            gsm.m.http_response.data[sizeof(gsm.m.http_response.data) - 1] = '\0';
```

Comment:
Is the -1 needed?  Usually if I have an array of size 16 and it's got a string of length 15 in it, I can add a null to the end with element 15, since the array is zero based.  The 15 character string will be in locations 0 - 14, and I can add a NULL at location 15.  I assume that this works, but just asking :)

Reply:
The strncpy function won't append the \0 if the source string is longer than n. This likely won't happen, but in case it does I added the manual setting of \0 at the end. This would overwrite whatever character is there but it's overflowed anyway so it doesn't matter and will prevent any garbage after the string from being included.

Reply:
Just reread and may have misinterpreted your question. The -1 is there because sizeof is the entire memory space which would include the NULL byte (unlike strlen which doesn't).